### PR TITLE
[kotlin-server][jax-rs] Added support for JAX-RS library into Kotlin Server generator

### PR DESCRIPTION
**Pull request by anttileppa**
Added support for JAX-RS library into Kotlin Server generator. New generator can be tested using kotlin-server generator with jaxrs-spec library.

Ping @jimschubert 

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
cc @jimschubert (2017/09) ❤️, @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04) @andrewemery (2019/10) @4brunu (2019/11) @yutaka0m (2020/03)

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinServerCodegen.java [hunk 1]
```diff
@@ -117,6 +128,11 @@ public KotlinServerCodegen() {
         addSwitch(Constants.COMPRESSION, Constants.COMPRESSION_DESC, getCompressionFeatureEnabled());
         addSwitch(Constants.LOCATIONS, Constants.LOCATIONS_DESC, getLocationsFeatureEnabled());
         addSwitch(Constants.METRICS, Constants.METRICS_DESC, getMetricsFeatureEnabled());
+
+        cliOptions.add(CliOption.newBoolean(INTERFACE_ONLY, "Whether to generate only API interface stubs without the server files.").defaultValue(String.valueOf(interfaceOnly)));
+        cliOptions.add(CliOption.newBoolean(USE_BEANVALIDATION, "Use BeanValidation API annotations",useBeanValidation));
+        cliOptions.add(CliOption.newBoolean(USE_COROUTINES, "Whether to use the Coroutines with the JAX-RS library."));
+        cliOptions.add(CliOption.newBoolean(RETURN_RESPONSE, "Whether generate API interface should return javax.ws.rs.core.Response instead of a deserialized entity. Only useful if interfaceOnly is true.").defaultValue(String.valueOf(returnResponse)));
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinServerCodegen.java:135]: @anttileppa thanks for the PR. Is it correct to say that these new options are for `jaxrs-spec` only?

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinServerCodegen.java]: Ok. Let me review again today.

**Comment by wing328** [comment]
```
BUILD SUCCESSFUL in 1m 57s
6 actionable tasks: 6 executed
```

Tested locally and the result is good. Thanks for the PR.

File: modules/openapi-generator/src/main/resources/kotlin-server/libraries/jaxrs-spec/api.mustache [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.Map
+import java.util.List
+{{#useBeanValidation}}import javax.validation.constraints.*
+import javax.validation.Valid{{/useBeanValidation}}
+
+{{#useSwaggerAnnotations}}
+@Api(description = "the {{{baseName}}} API"){{/useSwaggerAnnotations}}{{#hasConsumes}}
+@Consumes({ {{#consumes}}"{{{mediaType}}}"{{^-last}}, {{/-last}}{{/consumes}} }){{/hasConsumes}}{{#hasProduces}}
+@Produces({ {{#produces}}"{{{mediaType}}}"{{^-last}}, {{/-last}}{{/produces}} }){{/hasProduces}}
+@Path("/")
```
[the remark this answers is not in the document]
Comment (anttileppa) [reply modules/openapi-generator/src/main/resources/kotlin-server/libraries/jaxrs-spec/api.mustache:23]: @regispl sounds like a mistake to me

**Pull request merged**